pkg/resources/kafka/configmap.go

```diff
@@ -106,10 +112,24 @@ func (r *Reconciler) getConfigProperties(bConfig *v1beta1.BrokerConfig, id int32
 		log.Error(err, "setting broker.id in broker configuration resulted an error")
 	}
 
-	// Storage configuration
-	storageConf := generateStorageConfig(bConfig.StorageConfigs)
-	if storageConf != "" {
-		if err := config.Set("log.dirs", storageConf); err != nil {
+	// This logic prevents the removal of the mounthPath from the broker configmap
+	brokerConfigMapName := fmt.Sprintf(brokerConfigTemplate+"-%d", r.KafkaCluster.Name, id)
+	var brokerConfigMapOld v1.ConfigMap
+	err = r.Client.Get(context.Background(), client.ObjectKey{Name: brokerConfigMapName, Namespace: r.KafkaCluster.GetNamespace()}, &brokerConfigMapOld)
+	if err != nil && !apierrors.IsNotFound(err) {
+		log.Error(err, "getting broker configmap from the Kubernetes API server resulted an error")
+	}
+
+	mounthPathsOld := getMountPathsFromBrokerConfigMap(&brokerConfigMapOld)
```

Comment:
Good catch. It is only a log.Error :)

Reply:
It handles because (var brokerConfigMapOld v1.ConfigMap) this is not nil. It contains Data map[string]string. 
brokerConfig := configMap.Data[kafkautils.ConfigPropertyName]  here brokerConfig will be empty string, and return will be empty slice

Reply:
Im going to put a protection in it anyway.